target/linux/ath79/dts/ar9344_qxwlan_e750a-v4-8m.dts

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "ar9344_qxwlan_e750x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "Qxwlan E750A V4 8M";
+	compatible = "qxwlan,e750a-v4-8m", "qca,ar9344";
+};
+
+&leds {
+	lan {
+		label = "green:lan";
+		gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+	};
+
+	wan {
+		label = "green:wan";
+		gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+	};
+};
+
+&eth1 {
+	status = "okay";
+
+	mtd-mac-address = <&pridata 0x400>;
+
+	gmac-config {
+		device = <&gmac>;
+		switch-phy-swap = <0>;
+		switch-only-mode = <1>;
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	phy-handle = <&swphy4>;
+
+	mtd-mac-address = <&pridata 0x400>;
+	mtd-mac-address-increment = <1>;
+};
+
+&spi {
+	status = "okay";
+
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x040000 0x010000>;
+				read-only;
+			};
+
+			pridata: partition@50000 {
+				label = "pri-data";
+				reg = <0x050000 0x010000>;
+				read-only;
+			};
+
+			art: partition@60000 {
+				label = "art";
+				reg = <0x060000 0x010000>;
+				read-only;
+			};
+
+			partition@70000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x070000 0x790000>;
```

Comment:
Personally, I'd create another DTSI - ar9344_qxwlan_e750a-v4.dtsi - which would contain everything except the last partition and model/compatible. The DTS would then just be

```
// SPDX-License-Identifier: GPL-2.0-or-later OR MIT

#include "ar9344_qxwlan_e750a-v4.dtsi"

/ {
	model = "Qxwlan E750A V4 8M";
	compatible = "qxwlan,e750a-v4-8m", "qca,ar9344";
};

&partitions {
	partition@70000 {
		compatible = "denx,uimage";
		label = "firmware";
		reg = <0x070000 0x790000>;
	};
};
```

Reply:
(Note that this means _keeping_ target/linux/ath79/dts/ar9344_qxwlan_e750x.dtsi as well, the new DTSI would just contain the contents of e750a-v4-xm dtses)